### PR TITLE
Add apm_user reserved role

### PR DESCRIPTION
**Pull request by kobelb**
This adds a reserved role for APM to use to grant access to APM, as opposed to requiring users to create custom roles. This will allow us the flexibility to assign Kibana privileges to the reserved role as part of the [Kibana Feature Controls](https://github.com/elastic/kibana/issues/20277).

### CODE REVIEW

**Comment by makwarth** [comment]
@bmorelli25 This change will create a dedicated APM role in 7.0. This means that we'll need to update our docs to instruct people to use this role, as opposed to creating their own. Can you update the [docs](https://www.elastic.co/guide/en/apm/server/current/beats-user-access.html) accordingly?

@kobelb has offered to walk you through the changes if needed

**Comment by graphaelli** [comment]
Can the apm user also have read access to `.ml-anomalies*` by default?  @elastic/apm-ui can you confirm that's the only one needed from ml?  any others?

**Comment by kobelb** [comment]
> Can the apm user also have read access to .ml-anomalies* by default?

It definitely can be changed to do so, which version are you hoping to begin taking advantage of this?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 1]
```diff
@@ -132,6 +132,10 @@
                         new String[] { "monitor", MonitoringBulkAction.NAME}, null, null, MetadataUtils.DEFAULT_RESERVED_METADATA))
                 .put(UsernamesField.APM_ROLE, new RoleDescriptor(UsernamesField.APM_ROLE,
                         new String[] { "monitor", MonitoringBulkAction.NAME}, null, null, MetadataUtils.DEFAULT_RESERVED_METADATA))
+                .put("apm_user", new RoleDescriptor("apm_user",
+                    null, new RoleDescriptor.IndicesPrivileges[] {
+                        RoleDescriptor.IndicesPrivileges.builder().indices("apm-*").privileges("read", "view_index_metadata").build()
```
Comment (sqren) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java]: Users with Platinum also need access to ML anomaly indicies `.ml-anomalies-*`

**Comment by sqren** [comment]
> @elastic/apm-ui can you confirm that's the only one needed from ml? any others outside of ml

Yes, that's the only index we use outside `apm-*`

> It definitely can be changed to do so, which version are you hoping to begin taking advantage of this?

asap :) We are already reading from `.ml-anomalies-*` so we've had a few support tickets with this issue (where people have to create custom roles and access permissions to make it work).

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java (hunk 1 above)
Comment (sqren) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java]: (I see now that @graphaelli already pointed this out)

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 2]
```diff
@@ -134,7 +134,10 @@
                         new String[] { "monitor", MonitoringBulkAction.NAME}, null, null, MetadataUtils.DEFAULT_RESERVED_METADATA))
                 .put("apm_user", new RoleDescriptor("apm_user",
                     null, new RoleDescriptor.IndicesPrivileges[] {
-                        RoleDescriptor.IndicesPrivileges.builder().indices("apm-*").privileges("read", "view_index_metadata").build()
+                        RoleDescriptor.IndicesPrivileges.builder().indices("apm-*")
+                            .privileges("read", "view_index_metadata").build(),
+                        RoleDescriptor.IndicesPrivileges.builder().indices(".ml-anomalies*")
+                            .privileges("view_index_metadata", "read").build(),
```
Comment (sqren) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java]: 🙇

**Review by jaymode — APPROVED** [review]

**Comment by colings86** [comment]
Please remember to add all relevant labels (area label, version label(s) and change type label) on all PRs and please look for this as part of reviews. The release note generation process is made much harder when PRs are not labelled correctly.

**Comment by kobelb** [comment]
Sorry about that @colings86

**Pull request merged**